lldb/packages/Python/lldbsuite/test/builders/builder.py

```diff
@@ -96,16 +98,120 @@ def getArchSpec(self, architecture):
+            }
+            cxx_type = cxx_types.get(cc_type, cc_type)
+
+            cc_dir = cc_path.parent
+
+            def getLlvmUtil(util_name):
+                llvm_tools_dir = os.getenv("LLVM_TOOLS_DIR", cc_dir.as_posix())
+                return llvm_tools_dir + "/" + util_name + exe_ext
+
+            def getToolchainUtil(util_name):
+                return (cc_dir / (cc_prefix + util_name + cc_ext)).as_posix()
+
+            cxx = getToolchainUtil(cxx_type)
+
+            util_names = {
+                "OBJCOPY": "objcopy",
+                "STRIP": "objcopy",
+                "ARCHIVER": "ar",
+                "DWP": "dwp",
+            }
+            utils = []
+
+            # Note: LLVM_AR is currently required by API TestBSDArchives.py tests.
+            # Assembly a full path to llvm-ar for given LLVM_TOOLS_DIR;
+            # otherwise assume we have llvm-ar at the same place where is CC specified.
+            if not os.getenv("LLVM_AR"):
+                llvm_ar = getLlvmUtil("llvm-ar")
+                utils.extend(['LLVM_AR="%s"' % llvm_ar])
+
+            if not lldbplatformutil.platformIsDarwin():
+                if os.getenv("USE_LLVM_TOOLS"):
+                    # Use the llvm project tool instead of the system defaults.
+                    for var, name in util_names.items():
+                        utils.append('%s="%s"' % (var, getLlvmUtil("llvm-" + name)))
+                    utils.extend(['AR="%s"' % llvm_ar])
+                elif cc_type in ["clang", "cc", "gcc"]:
+                    util_paths = {}
+                    # Assembly a toolchain side tool cmd based on passed CC.
+                    for var, name in util_names.items():
+                        # Do not override explicity specified tool from the cmd line.
+                        if not os.getenv(var):
+                            util_paths[var] = getToolchainUtil(name)
+                        else:
+                            util_paths[var] = os.getenv(var)
+                    utils.extend(['AR="%s"' % util_paths["ARCHIVER"]])
+
+                    # Look for llvm-dwp or gnu dwp
+                    if not lldbutil.which(util_paths["DWP"]):
+                        util_paths["DWP"] = getToolchainUtil("llvm-dwp")
+                    if not lldbutil.which(util_paths["DWP"]):
+                        util_paths["DWP"] = lldbutil.which("llvm-dwp")
+                    if not util_paths["DWP"]:
+                        util_paths["DWP"] = lldbutil.which("dwp")
+                        if not util_paths["DWP"]:
+                            del util_paths["DWP"]
+
+                    for var, path in util_paths.items():
+                        utils.append('%s="%s"' % (var, path))
+            else:
+                utils.extend(['AR="%slibtool"' % os.getenv("CROSS_COMPILE", "")])
+
+            return [
+                'CC="%s"' % cc,
+                'CCC="%s"' % cc_type,
+                'CXX="%s"' % cxx,
```

Comment:
I.e., by replacing this (and others) with
```suggestion
                'CC=%s' % cc,
                'CCC=%s' % cc_type,
                'CXX=%s' % cxx,
```